server/src/main/java/org/apache/druid/discovery/DiscoveryDruidNode.java

```diff
@@ -80,6 +84,16 @@ public DruidNode getDruidNode()
     return druidNode;
   }
 
+  public boolean isDiscoverableDataServer()
+  {
+    final DruidService druidService = services.get(DataNodeService.DISCOVERY_SERVICE_KEY);
+    if (druidService == null) {
+      return false;
+    }
+    final DataNodeService dataNodeService = (DataNodeService) druidService;
+    return dataNodeService.isDiscoverable();
+  }
+
   public DruidServer toDruidServer()
   {
     return new DruidServer(
```

Comment:
should this precondition check `isDiscoverableDataServer`?

Reply:
Yes, it should. Looking at this and `isDiscoverableDataServer` closely, it seems better to have them as util methods. I moved them into static methods in `ServersTable`.